flow/record/fieldtypes/__init__.py

```diff
@@ -600,12 +601,18 @@ def _unpack(cls, data):
         return data
 
 
-def _is_posixlike_path(path: Any):
-    return isinstance(path, pathlib.PurePath) and "\\" not in (path._flavour.sep, path._flavour.altsep)
+def _is_posixlike_path(path: Any) -> bool:
+    if isinstance(path, pathlib.PurePath):
+        obj = getattr(path, "parser", None) or path._flavour
```

Comment:
Would this also work?

```suggestion
        obj = getattr(path, "parser", path._flavour)
```

Reply:
No, since `._flavour` will result in an attribute error on newer Python versions because it'll get eagerly executed to pass the result to `getattr`. In this syntax, `._flavour` only gets executed in case `parser` does not exist (happy path for the future).